libs/types/src/tokens.rs

```diff
@@ -150,4 +174,7 @@ pub struct CustomMetadata {
 
 	/// Whether an asset can be used as a currency to fund Centrifuge Pools.
 	pub pool_currency: bool,
+	// TODO: Enable in follow-up PR
+	// /// The corresponding 20-byte EVM address of the asset.
+	// pub evm_address: [u8; 20],
```

Comment:
This won't work because this would assume that every token would have to have an `evm_address`. We could either make it optional, which is still a weak abstraction in my opinion, or we can use the `MultiLocation` field of the metadata to represent the EVM address of that token, assuming we are registering the asset as it being an asset in a Connector-connected Domain, if that makes sense? By this last part I mean, lets's say that Domain X has a Token `X` with an EVM address `EA`; if we register `X` on the `Centrifuge Chain` just to register/list a token we handle on Domain `X`, that approach could work, because when sending/receiving transfers from/to that Domain, we can expect the respective entry in the asset registry to have a MultiLocation holding an EVM address; if it doesn't, we throw an error.

Maybe I am making this sound too complicated, but I am coming from a perspective where every token listed in the asset registry (apart from Tranche tokens) is a potential token that could be transferred through XCM; by using `MultiLocation` to hold an `EVM` address that wouldn't fit _that_ purpose, but since that's not the purpose we would be registering the token in the registry in the first place, I think that's fine.

WDYT?

Reply:
I definitely get your point! This is purely based on the [last ConnectorsV2 sync](https://centrifuge.hackmd.io/ODJ-dNHUSsWcYNh6dHJtcg?both#Outcome) we had were we decided to add `evm_address` to the metadata of currencies in the asset registry. I am not sure though how to deterministically derive 32 bytes from the flexible `MultiLocation` which could be quite scaffolded, e.g. `Here` vs. `X2(para, pallet_instance, general_index)` etc. How would you handle that?

Reply:
We could use a `X2(DomainId, AccountKey20 { network: NetworkId, key: [u8; 20] }}`; but that's probably trying to make a shoe out of a shoe box. I remember @offerijns suggesting in that sync to maybe use a type like `DomainAddress`; the issue with that is that we already have the `location` field on the `Metadata` type. 

As I am thinking and writing about this, it becomes clearer how there's an abstraction leakage here, in the sense of trying to use a design / abstraction (the asset registry) for something it's not quite designed for. The best would be if we could change the `Meadata.location` field to be something like `enum GlobalLocation { Polkadot(VersionedMultiLocation), EVM(chainId, address) }` but we can't do that unless if we change the orml pallet itself.

I have to think a bit more about this 🙃

Reply:
Why not use XCM MultiLocations for this, weren't they explicitly designed to support multiple consensus systems, so also able to represent EVM addresses? We can/should look at how Snowbridge intents to do this.